### PR TITLE
Refactor statdb

### PR DESCRIPTION
**Pull request by mobyvb**
https://storjlabs.atlassian.net/browse/V3-837

* Update statdb args/return values to minimize structs
* Simplify statdb.Update() to update all stats instead of an arbitrary subset determined by flags
* Remove CreateIfNotExists logic from statdb.Update()
* Simplify audit code structure

### CODE REVIEW

File: satellite/satellitedb/statdb.go [hunk 1]
```diff
@@ -31,7 +31,8 @@ type statDB struct {
 }
 
 // Create a db entry for the provided storagenode
-func (s *statDB) Create(ctx context.Context, createReq *statdb.CreateRequest) (resp *statdb.CreateResponse, err error) {
+func (s *statDB) Create(ctx context.Context, nodeID storj.NodeID,
+	startingStats *pb.NodeStats) (stats *pb.NodeStats, err error) {
```
Comment (egonelbre) [inline satellite/satellitedb/statdb.go]: nit: single line

File: pkg/audit/reporter.go [hunk 2]
```diff
@@ -22,6 +22,13 @@ type Reporter struct {
 	maxRetries int
 }
 
+// RecordAuditsInfo is a struct containing arguments/return values for RecordAudits()
+type RecordAuditsInfo struct {
```
Comment (egonelbre) [inline pkg/audit/reporter.go]: Let's also use a separate type for `NodeStats` and add a converter if necessary.

File: pkg/statdb/statdb_test.go [hunk 3]
```diff
@@ -17,9 +17,9 @@ import (
 	"storj.io/storj/satellite/satellitedb/satellitedbtest"
 )
 
-func getRatio(s, t int) (success, total int64, ratio float64) {
+func getRatio(s, t int64) (success, total int64, ratio float64) {
```
Comment (egonelbre) [inline pkg/statdb/statdb_test.go]: nit: let's use longer names instead of `s`, `t`...

File: pkg/statdb/statdb_test.go [hunk 4]
```diff
@@ -45,34 +50,28 @@ func testDatabase(ctx context.Context, t *testing.T, sdb statdb.DB) {
...
 		assert.NoError(t, err)
-		s := resp.Stats
 		assert.EqualValues(t, auditSuccessRatio, s.AuditSuccessRatio)
 		assert.EqualValues(t, uptimeRatio, s.UptimeRatio)
 
-		getReq := &statdb.GetRequest{
-			Node: nodeID,
-		}
-		getResp, err := sdb.Get(ctx, getReq)
+		s, err = sdb.Get(ctx, nodeID)
```
Comment (egonelbre) [inline pkg/statdb/statdb_test.go]: nit: let's use longer names instead of `s`..., e.g. `stats`

File: pkg/statdb/statdb_test.go [hunk 5]
```diff
@@ -125,131 +117,111 @@ func testDatabase(ctx context.Context, t *testing.T, sdb statdb.DB) {
...
-		auditSuccessCount, auditCount, auditSuccessRatio := getRatio(4, 10)
-		uptimeSuccessCount, uptimeCount, uptimeRatio := getRatio(8, 25)
+		auditSuccessCount, auditCount, auditSuccessRatio := getRatio(currAuditSuccess, currAuditCount)
+		uptimeSuccessCount, uptimeCount, uptimeRatio := getRatio(currUptimeSuccess, currUptimeCount)
 
-		getReq := &statdb.GetRequest{
-			Node: nodeID,
-		}
-		getResp, err := sdb.Get(ctx, getReq)
+		s, err := sdb.Get(ctx, nodeID)
```
Comment (egonelbre) [inline pkg/statdb/statdb_test.go]: nit: let's use longer names instead of `s`..., e.g. `stats`

File: pkg/statdb/statdb_test.go [hunk 6]
```diff
@@ -125,131 +117,111 @@ func testDatabase(ctx context.Context, t *testing.T, sdb statdb.DB) {
...
-		auditSuccessCount, auditCount, auditSuccessRatio := getRatio(5, 11)
-		uptimeSuccessCount, uptimeCount, uptimeRatio := getRatio(8, 26)
+		auditSuccessCount, auditCount, auditSuccessRatio := getRatio(currAuditSuccess, currAuditCount)
+		uptimeSuccessCount, uptimeCount, uptimeRatio := getRatio(currUptimeSuccess, currUptimeCount)
 
-		getReq := &statdb.GetRequest{
-			Node: nodeID,
-		}
-		getResp, err := sdb.Get(ctx, getReq)
+		s, err := sdb.Get(ctx, nodeID)
```
Comment (egonelbre) [inline pkg/statdb/statdb_test.go]: nit: let's use longer names instead of `s`..., e.g. `stats`

File: pkg/statdb/statdb_test.go [hunk 7]
```diff
@@ -269,13 +241,9 @@ func testDatabase(ctx context.Context, t *testing.T, sdb statdb.DB) {
 			UptimeCount:        uptimeCount1,
 			UptimeRatio:        uptimeRatio1,
 		}
-		createReq := &statdb.CreateRequest{
-			Node:  nodeID1,
-			Stats: nodeStats,
-		}
-		resp, err := sdb.Create(ctx, createReq)
+
+		s, err := sdb.Create(ctx, nodeID1, nodeStats)
```
Comment (egonelbre) [inline pkg/statdb/statdb_test.go]: nit: let's use longer names instead of `s`..., e.g. `stats`

File: pkg/audit/reporter.go (hunk 2 above)
Comment (coyle) [reply pkg/audit/reporter.go]: curious as to the reason why

Comment (egonelbre) [reply pkg/audit/reporter.go]: Separating the layers. protobuf is an implementation detail not the core data-structures of the system. Using protobuf in the underlying layers makes codes somewhat harder to test and read.

When you look at the layers there are:

```
database <- ... <- service <- endpoint <- transport
```

protobuf, grpc etc. are all about transport, not the protocol semantics. By depending directly on protobuf this separation less clearer and hence the whole structure less clearer.

**Pull request merged**